### PR TITLE
Standardize CI 20200203

### PR DESCRIPTION
**Pull request by phil-davis**
- cleanup `.gitignore`
- implement Travis like in `skel` (separate `phpstan` job, `--prefer-lowest` in matrix)
- apply `php-cs-fixer` changes for `tests` folder
- move tests into `tests/Event` and add composer `autoload-dev` section to allow `phpstan` to analyze the test code
- refactor the `BenchMark` tests into 1 file per class so that `phpstan` can analyze them.

### CODE REVIEW

**Comment by phil-davis** [comment]
@staabm the "BenchMark" "tests" had to be re-arranged to keep `phpstan` happy.
Actually they are not `phpunit` tests and are not run by `phpunit`.
Is the refactoring OK with you?
Or what to do?

File: tests/Event/Loop/LoopTest.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ public function testTick()
             ++$check;
         }, 1);
 
-        $loop->nextTick(function () use ($loop, &$check) {
+        $loop->nextTick(function () use (&$check) {
```
Comment (phil-davis) [inline tests/Event/Loop/LoopTest.php]: `phpstan` reported that `$loop` was unused

**Comment by phil-davis** [comment]
I added a commit "Use at least current phpunit minor versions"
As was done in `skel` https://github.com/sabre-io/skel/pull/12

File: .travis.yml [hunk 2]
```diff
@@ -10,27 +10,23 @@ env:
...
-  - composer update --prefer-source $LOWEST_DEPS
+  - composer update --prefer-source $PREFER_LOWEST
 
 script:
-  - if [ $RUN_PHPSTAN == "FALSE"  ]; then ./bin/php-cs-fixer  fix --dry-run --diff; fi
-  - if [ $RUN_PHPSTAN == "FALSE"  ]; then ./bin/phpunit --configuration tests/phpunit.xml --coverage-clover=coverage.xml; fi
-  - if [ $RUN_PHPSTAN == "TRUE"  ]; then ./bin/phpstan.phar analyse -c phpstan.neon lib; fi
+  - if [ $RUN_PHPSTAN == "FALSE" ]; then php vendor/bin/php-cs-fixer fix --dry-run --diff; fi
+  - if [ $RUN_PHPSTAN == "FALSE" ]; then php vendor/bin/phpunit --configuration tests/phpunit.xml $WITH_COVERAGE; fi
+  - if [ $RUN_PHPSTAN == "TRUE" ]; then php vendor/bin/phpstan analyse -c phpstan.neon lib tests; fi
```
Comment (staabm) [inline .travis.yml]: in case we decide it on the other PR: this paths could be part of the phpstan.neon config-file

Comment (phil-davis) [reply .travis.yml]: yes, if we decide to change the "template" `phpstan.neon` then we can make another PR to each repo.

**Pull request merged**